### PR TITLE
costmodel update for zhaohang project

### PR DESCRIPTION
**Pull request by alishenli**

### CODE REVIEW

File: lib/transforms/analyzer.cc [hunk 1]
```diff
@@ -531,12 +641,90 @@ void Analyzer::WriteCSVReport(std::ostream& os) {
...
+  float init_latency = static_cast<float>(conv_op_num) * conv_knl_init;
+  init_latency += static_cast<float>(matmul_op_num) * mm_knl_init;
+  init_latency +=
+      static_cast<float>(other_op_num - conv_act_num) * other_knl_init;
+  float knl_latency = conv_time * (conv_flops + conv_act_flops);
+  knl_latency += mm_time * matmul_flops;
+  knl_latency +=
+      other_time * (total_flops - conv_flops - conv_act_flops - matmul_flops);
+
+  const float t4_max_mem = 16000; // MB
```
Comment (weimingzha0) [inline lib/transforms/analyzer.cc:667]: max_mem ?

Comment (alishenli) [reply lib/transforms/analyzer.cc]: put it into the hw_info map

File: include/halo/halo.h [hunk 2]
```diff
@@ -53,6 +53,7 @@ enum class ChannelOrder {
 struct AnalyzerOpts {
   bool print_details = false;
   int batch_size = 1;
+  int ips = 0; // image per second
```
[the remark this answers is not in the document]
Comment (alishenli) [reply include/halo/halo.h]: done

File: include/halo/lib/transforms/analyzer.h [hunk 3]
```diff
@@ -65,12 +66,27 @@ class Analyzer final : public ModulePass {
     size_t size = 0;
   };
 
+  struct HW_Info {
```
Comment (weimingzha0) [inline include/halo/lib/transforms/analyzer.h:69]: Use "HWInfo"? so it's more consistent with other type names like "TensorInfo"

**Review by weimingzha0 — COMMENTED** [review]
just a minor naming convention. otherwise, it's good.

File: include/halo/lib/transforms/analyzer.h (hunk 3 above)
Comment (alishenli) [reply include/halo/lib/transforms/analyzer.h]: Sure, good catch! Updated.

**Review by weimingzha0 — APPROVED** [review]

**Pull request merged**